modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackService.kt

```diff
@@ -651,4 +675,67 @@ open class PlaybackService : MediaBrowserServiceCompat(), CoroutineScope {
         // convert podcasts to the media browser format
         return podcasts.mapNotNull { podcast -> convertPodcastToMediaItem(context = this, podcast = podcast, useEpisodeArtwork = settings.artworkConfiguration.value.useEpisodeArtwork) }
     }
+
+    private fun observePlaybackState() {
+        sleepTimer.stateFlow
+            .map { it }
+            .distinctUntilChanged()
+            .flowOn(Dispatchers.IO)
+            .onEach { state ->
+                onSleepTimerStateChange(state)
+            }
+            .catch { throwable ->
+                Timber.e(throwable, "Error observing SleepTimer state")
+            }
+            .launchIn(this)
+    }
```

Comment:
I was thinking to observe the sleep timer changes by using `sleeptimer.stateFlow.collect {}` instead, but searching for the best approach I found this one. 

The `collect { }` is suspending, which means it will block the current coroutine until the flow is canceled.